Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -38,13 +38,17 @@ public struct HTTPRequestHead: Equatable {
     /// The header fields for this HTTP request.
     public var headers: HTTPHeaders
 
+    /// The keepAlive value that was detected by http_parser. This can only be used safely
+    /// as long as the user has not made modifications to this header block.
+    private let keepAlive: Bool?
```

Comment:
I don't really like having a state variable here that is only valid when also consulting another variable. I'd say the whole state should live in the headers. Couldn't we architect it this way: We allow (as an `internal` constructor) to pre-seed the headers with a keep alive state. Whenever the headers change, we invalidate that.

So the keep-alive state in the headers could be one of the following:

- `keepAlive`: no header scan needed, I know I have a `Connection: keep-alive` header
- `close`: no header scan needed, I know I have a `Connection: close` header
- `default`: no header scan needed, I know I had neither `Connection: keep-alive` nor `Connection: close`
- `needsScan`: header scan needed

so internally we could pre-seed it with either `keepAlive`, `close`, or `default`, depending on what http_parser tells us. Then on any modification we change it to `needsScan`.